webapp/src/ts/modules/contacts/contacts-report.component.ts

```diff
@@ -0,0 +1,241 @@
+import { TranslateService } from '@ngx-translate/core';
+import { isEqual as _isEqual } from 'lodash-es';
+
+import { ContactViewModelGeneratorService } from '@mm-services/contact-view-model-generator.service';
+import { EnketoService } from '@mm-services/enketo.service';
+import { GeolocationService } from '@mm-services/geolocation.service';
+import { GlobalActions } from '@mm-actions/global';
+import { Selectors } from '@mm-selectors/index';
+import { TelemetryService } from '@mm-services/telemetry.service';
+import { TranslateFromService } from '@mm-services/translate-from.service';
+import { XmlFormsService } from '@mm-services/xml-forms.service';
+
+@Component({
+  templateUrl: './contacts-report.component.html'
+})
+export class ContactsReportComponent implements OnInit, OnDestroy, AfterViewInit {
+  private globalActions;
+  private geoHandle:any;
+  private routeSnapshot;
+  private telemetryData:any = {
+    preRender: Date.now()
+  };
+
+  subscription: Subscription = new Subscription();
+  enketoEdited
+  enketoStatus;
+  enketoSaving;
+  enketoError;
+  selectedContact;
+  form;
+  loadingForm;
+  errorTranslationKey;
+  contentError;
+  cancelCallback;
+
+  constructor(
+    private store: Store,
+    private enketoService: EnketoService,
+    private geolocationService: GeolocationService,
+    private telemetryService: TelemetryService,
+    private xmlFormsService: XmlFormsService,
+    private translateFromService: TranslateFromService,
+    private router: Router,
+    private route: ActivatedRoute,
+    private translateService: TranslateService,
+    private contactViewModelGeneratorService: ContactViewModelGeneratorService,
+  ){
+    this.globalActions = new GlobalActions(store);
+  }
+
+  ngOnInit() {
+    this.subscribeToStore();
+    this.subscribeToRoute();
+
+    this.geoHandle = this.geolocationService.init();
+    this.resetFormError();
+    this.form = null;
+    this.loadingForm = true;
+    this.globalActions.setShowContent(true);
+    this.setCancelCallback();
+    this.globalActions.clearRightActionBar();
+  }
+
+  ngOnDestroy() {
+    this.subscription.unsubscribe();
+    this.geoHandle && this.geoHandle.cancel();
+    this.enketoService.unload(this.form);
+  }
+
+  ngAfterViewInit() {
+    this.render();
+  }
+
+  private getContact() {
+    const id = this.routeSnapshot.params.id;
+    if (!id) {
+      return Promise.resolve();
+    }
+    return this.contactViewModelGeneratorService
+      .getContact(id, { getChildPlaces: false, merge: true })
+      .then((result) => result.doc);
+  }
+
+  private resetState() {
+    this.loadingForm = true;
+
+    if (!this.routeSnapshot.params?.id) {
+      this.globalActions.unsetSelected();
+      this.globalActions.settingSelected();
+    }
+  }
+
+  private markFormEdited() {
+    this.globalActions.setEnketoEditedStatus(true);
+  }
+
+  private resetFormError() {
+    if (this.enketoError) {
+      this.globalActions.setEnketoError.setEnketoError(null);
```

Comment:
This line is causing the error described above. 
Please add a unit test for this.

Reply:
Good catch. Fixed.